core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -319,6 +322,9 @@ public int from() {
      * The number of search hits to return. Defaults to <tt>10</tt>.
      */
     public SearchSourceBuilder size(int size) {
+        if (size < 0) {
+            throw new IllegalArgumentException("size must be no negative but was [" + size + "]");
```

Comment:
A better wording for the error is: `[size] can't be negative`
Changing that will require updating tests